### PR TITLE
Add undeploy cmd

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
Implement undeploy command to fix cases where the deletion gets stuck because there are traces still running. 

Fixes https://github.com/kinvolk/inspektor-gadget/issues/388 
Ref https://github.com/kinvolk/inspektor-gadget/issues/123

TODO:
- [x] Update once is https://github.com/kinvolk/inspektor-gadget/pull/429 merged. 

TODO next PRs:
- [ ] Improvements for the deploy command (https://github.com/kinvolk/inspektor-gadget/issues/123) are future work.

### CODE REVIEW

**Review by eiffel-fl — APPROVED** [review]
This contribution is welcomed! Thank you for it!
I tested it and it is OK.

I just found two nits, also can you please update your commit message to `s/trickly/tricky/`?
Also, can you please update `Makefile` to replace this:

```bash
./kubectl-gadget deploy | kubectl delete -f - || true
```

File: cmd/kubectl-gadget/undeploy.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+		// It's taking too long to delete the traces. Remove the
+		// finalizers and let k8s remove them immediately.
+		for _, trace := range traces.Items {
+			data := []byte("{\"metadata\":{\"finalizers\":[]}}")
+			_, err := traceClient.GadgetV1alpha1().Traces("gadget").Patch(
+				context.TODO(), trace.Name, types.MergePatchType, data, metav1.PatchOptions{},
+			)
+			if err != nil {
+				errs = append(errs, fmt.Sprintf("failed to patch the traces: %s", err))
```
[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply cmd/kubectl-gadget/undeploy.go:106]: Seems like a good idea. Fixed.

File: cmd/kubectl-gadget/undeploy.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		i++
+		if i < n {
+			delay = 2 * delay
+			goto again
+		}
+
+		// It's taking too long to delete the traces. Remove the
+		// finalizers and let k8s remove them immediately.
+		for _, trace := range traces.Items {
+			data := []byte("{\"metadata\":{\"finalizers\":[]}}")
```
[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply cmd/kubectl-gadget/undeploy.go:101]: Done!

**Review by eiffel-fl — APPROVED** [review]
LGTM!

It will be perfect once you addressed the nit and can you please also apply this patch:

```patch
diff --git a/integration/command.go b/integration/command.go
index b85c166..04a98dd 100644
--- a/integration/command.go
+++ b/integration/command.go
@@ -104,10 +104,9 @@ var cleanupTestNamespace *command = &command{
 }
 
 var cleanupInspektorGadget *command = &command{
-       name:           "cleanup gadget deployment",
-       cmd:            "$KUBECTL_GADGET deploy $GADGET_IMAGE_FLAG | kubectl delete --wait=false -f -",
-       expectedRegexp: "\"gadget\" deleted",
-       cleanup:        true,
+       name:    "cleanup gadget deployment",
+       cmd:     "$KUBECTL_GADGET undeploy",
+       cleanup: true,
 }
 
 // run runs the command on the given as parameter test.
```

File: Makefile [hunk 3]
```diff
@@ -164,7 +164,7 @@ minikube-install: gadget-container kubectl-gadget
 	# Delete traces CRD first: the gadget DaemonSet needs to be running
 	# because of Finalizers.
 	kubectl delete crd traces.gadget.kinvolk.io || true
-	./kubectl-gadget deploy | kubectl delete -f - || true
+	./kubectl-gadget undeploy || true
 	time kubectl wait --for=delete namespace gadget 2>/dev/null || true
 	time kubectl wait --for=delete daemonset -n kube-system gadget 2>/dev/null || true
```
[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply Makefile:169]: undeploy doesn't wait for the commands to complete, hence those two lines are still needed.

[the remark this answers is not in the document]
Comment (eiffel-fl) [reply Makefile:169]: Good catch!

[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply Makefile:167]: Yes yes, I missed it. Updated to use only `undeploy`.

[the remark this answers is not in the document]
Comment (eiffel-fl) [reply Makefile:167]: No problem, the `Makefile` is not really easy to read.

**Pull request merged**